driver/internal/protocol/session.go

```diff
@@ -58,9 +59,19 @@ func NewSession(ctx context.Context, rw *bufio.ReadWriter, cfg *SessionConfig) (
 
 	s := &Session{cfg: cfg, sessionID: defaultSessionID, pr: pr, pw: pw}
 
-	authStepper := newAuth(cfg.Username, cfg.Password)
+	var stepper authStepper
+	if cfg.Username != "" {
+		stepper = newAuth(cfg.Username, cfg.Password)
+	} else if cfg.Password != "" {
```

Comment:
this is how the JWT authentication method is integrated.
You should check if you're fine with it.

Reply:
I would recommend to also check if the password starts with "ey". As this is always the beginning of a valid JWT token.
Later on the SAML check will do the same, but it will check if the password begins with "<".